### ISSUE TITLE
Variable assignments in for/each loops

### ISSUE THREAD

**Issue by adrogon** [neuland/jade4j#153]
Hello, please consider the following dummy code:
```
- var books = ["A", "B", "C"]
- var foo = false
each book in books
  - foo = true

p= foo
```
According to your [interactive jade documentation](http://naltatis.github.io/jade-syntax-docs/), the expected ouput is: `<p>true</p>`
Although, the received output in Jade4J is: `<p>false</p>`

I know JEXL is not quite the same as what your interactive documentation uses, so what would be the correct piece of code to retrieve `true`?

Thank you.

PS: I worked around it by injecting other variables in Java, or using Helper classes, but I'd rather have it work in the template files if it is supposed to.

**Comment by dawez** [comment]
I tried as well and stepping in the source code, there is the class EachNode.java. In this class when executing a block inside a foreach, the following code is executed:

```
@Override
	public void execute(IndentWriter writer, JadeModel model, JadeTemplate template) throws JadeCompilerException {
		Object result;
		try {
			result = template.getExpressionHandler().evaluateExpression(getCode(), model);
		} catch (ExpressionException e) {
			throw new JadeCompilerException(this, template.getTemplateLoader(), e);
		}
		if (result == null) {
			throw new JadeCompilerException(this, template.getTemplateLoader(), "[" + code + "] has to be iterable but was null");
		}
		model.pushScope();
		run(writer, model, result, template);
		model.popScope();
	}
```
This causes the code executed inside the block in the foreach to be part of another scope.

**Comment by kchuykov** [comment]
will this issue be fixed?

**Comment by naltatis** [comment]
This is not really a bug. Introducing block scopes in several places was a design decision to avoid leaking variables.
One could also argue that collecting state in a render loop and using it for later elements is a code smell. It’s doing logic in the template. Granted jade is not a pure logicless template language by design, but we recommend using it as such.
For alignment with our primary goal, being compatible with the JS implementation, there is also a valid point an allowing this behavior to bridge the gap.

**Comment by adrogon** [comment]
So, is that a... yes? :smirk:

**Comment by dawez** [comment]
@naltatis You are right. My expectations would be that the template should behave as it is behaving in the JS implementation.

**Comment by chbloemer** [comment]
Will be fixed in 1.2.8
